bundle/compliance/lib/output_validations/test.rego

```diff
@@ -1,441 +1,39 @@
-	}
-
-	not validate_k8s_metadata(invalid_metadata)
-}
-
-test_validate_k8s_metadata_invalid_references {
-	invalid_metadata := {
-		"id": "rule id",
-		"name": "rule name",
-		"profile_applicability": "rule profile_applicability",
-		"description": "rule description",
-		"rationale": "rule rationale",
-		"audit": "rule audit",
-		"remediation": "rule remidiation",
-		"impact": "rule impact",
-		"default_value": "rule default_value",
-		"References": "rule references", # <- capitalized. should be "references"
-		"section": "rule section",
-		"version": "rule version",
-		"tags": ["tag 1", "tag 2"],
-		"benchmark": {"name": "benchmark", "version": "v1.0.0"},
-	}
-
-	not validate_k8s_metadata(invalid_metadata)
-}
-
-test_validate_k8s_metadata_invalid_section {
-	invalid_metadata := {
-		"id": "rule id",
-		"name": "rule name",
-		"profile_applicability": "rule profile_applicability",
-		"description": "rule description",
-		"rationale": "rule rationale",
-		"audit": "rule audit",
-		"remediation": "rule remidiation",
-		"impact": "rule impact",
-		"default_value": "rule default_value",
-		"references": "rule references",
-		"Section": "rule section", # <- capitalized. should be "section"
-		"version": "rule version",
-		"tags": ["tag 1", "tag 2"],
-		"benchmark": {"name": "benchmark", "version": "v1.0.0"},
-	}
-
-	not validate_k8s_metadata(invalid_metadata)
-}
-
-test_validate_k8s_metadata_invalid_version {
-	invalid_metadata := {
-		"id": "rule id",
-		"name": "rule name",
-		"profile_applicability": "rule profile_applicability",
-		"description": "rule description",
-		"rationale": "rule rationale",
-		"audit": "rule audit",
-		"remediation": "rule remidiation",
-		"impact": "rule impact",
-		"default_value": "rule default_value",
-		"references": "rule references",
-		"section": "rule section",
-		"Version": "rule version", # <- capitalized. should be "version"
-		"tags": ["tag 1", "tag 2"],
-		"benchmark": {"name": "benchmark", "version": "v1.0.0"},
+import future.keywords.every
+
+k8s_valid_metadata := {
+	"id": "rule id",
+	"name": "rule name",
+	"profile_applicability": "rule profile_applicability",
+	"description": "rule description",
+	"rationale": "rule rationale",
+	"audit": "rule audit",
+	"remediation": "rule remidiation",
+	"impact": "rule impact",
+	"default_value": "rule default_value",
+	"references": "rule references",
+	"section": "rule section",
+	"version": "rule version",
+	"tags": ["tag 1", "tag 2"],
+	"benchmark": {"name": "benchmark", "version": "v1.0.0", "id": "cis_k8s"},
+}
+
+eks_valid_metadata := {
+	"name": "rule name",
+	"description": "rule description",
+	"impact": "rule impact",
+	"tags": ["tag 1", "tag 2"],
+	"benchmark": {"name": "benchmark", "version": "v1.0.0", "id": "cis_eks"},
+	"remediation": "rule remidiation",
+}
+
+test_required_metadata_fields {
+	every key, _ in k8s_valid_metadata {
+		not validate_k8s_metadata(object.remove(k8s_valid_metadata, [key]))
```

Comment:
nice, however it doesn't check for nested fields like `benchmark.name` or `benchmark.id`
instead it removes the entire `benchmark` field

Reply:
cc @oren-zohar

Reply:
I agree. I thinks we should live with the current solution, once we have the same data validations for both EKS and Vanilla it would be easier to implement a more robust test.